src/js/static-filtering-parser.js

```diff
@@ -2448,10 +2448,9 @@ const NetOptionsIterator = class {
                 assigned &&
                     hasNoBits(descriptor, OPTMayAssign | OPTMustAssign) ||
                 assigned === false &&
-                    hasBits(descriptor, OPTMustAssign) && (
```

Comment:
See https://github.com/gorhill/uBlock/pull/3800/files#r681719585

Reply:
Maybe this big conditional expression should be just be re-written as method:

```js
optionIsValidInContext(descriptor, negated, assigned) {
    if ( descriptor === undefined ) {
        return false;
    }
    if ( negated && hasNoBits(descriptor, OPTCanNegate) )  {
        return false;
    }
    if ( this.exception && hasBits(descriptor, OPTBlockOnly) ) {
        return false;
    }
    if ( this.exception === false && hasBits(descriptor, OPTAllowOnly) {
        return false;
    }
    if ( assigned && hasNoBits(descriptor, OPTMayAssign | OPTMustAssign) ) {
        return false;
    }
    if ( assigned === false && hasBits(descriptor, OPTMustAssign) ) {
        if ( this.exception === false || hasNoBits(descriptor, OPTAllowMayAssign) ) {
            return false;
        }
    }
    return true;
}
```
This gets rid of indent issue, hopefully with no negative performance impact.

Reply:
This is not the only one, but I guess we can try to remove the exception for `LogicalExpression` later and do it as part of that work.

Reply:
Follow-up: https://github.com/gorhill/uBlock/pull/3801